ldms/src/ldmsd/ldmsd_stream.c

```diff
@@ -650,6 +650,7 @@ int ldmsd_stream_publish_file(const char *stream, const char *type,
 		return ENOMEM;
 	}
 
+	char * b = NULL;
```

Comment:
@baallan, I'm guessing the tool complained because there are goto's (goto err;) above the original declaration and the code at err: calls free(b).

That's fine, but that's not why this code is broken. If we go down to line 719 - 724, you'll see that we malloc(max_msg_len) and then call fgets(..., sizeof(b) -1, ...)

Well that sizeof(b)-1 is 7 bytes, not max_msg_len - 1.

So that's going to read our message 7B at a time and each time 7B is too small, it will realloc the destination buffer * 2 because it _thinks_ it's failing to read max_msg_len bytes, not 7 bytes.

Can you fix this please?

Reply:
@nichamon thanks for #1940